lisa/operating_system.py

```diff
@@ -649,6 +649,30 @@ def get_repositories(self) -> List[RepositoryInfo]:
 
         return repositories
 
+    def add_repository(
+        self, url: str, codename: str, packages: str = "main", key_location: str = ""
+    ) -> None:
+        if key_location:
+            wget = self._node.tools[Wget]
+            key_file_path = wget.get(
+                url=key_location,
+                file_path=str(self._node.working_path),
+            )
+            self._node.execute(
+                cmd=f"apt-key add {key_file_path}",
+                sudo=True,
+                expected_exit_code=0,
+                expected_exit_code_failure_message="fail to add apt key",
+            )
+        # This command will trigger apt update too, so it doesn't need to update
+        # repos again.
+        self._node.execute(
+            cmd=f'apt-add-repository "deb {url} {codename} {packages}"',
```

Comment:
add "-y"?

Reply:
which condition does it need? The current command works without it. Let me know, if there are some cases doesn't work, I will fix it in next PR.